src/handler.rs

```diff
@@ -176,6 +184,24 @@ impl RequestHandler {
                 return Ok(resp);
             }
 
+            // Metrics endpoint check
+            if metrics_request {
+                let body = if method.is_get() {
+                    let encoder = prometheus::TextEncoder::new();
+                    let mut buffer = Vec::new();
+                    encoder
+                        .encode(&prometheus::default_registry().gather(), &mut buffer)
+                        .unwrap();
+                    let data = String::from_utf8(buffer.clone()).unwrap();
+
+                    Body::from(data)
+                } else {
+                    Body::empty()
+                };
+                let resp = Response::new(body);
+                return Ok(resp);
+            }
+
             // Reject in case of incoming HTTP request method is not allowed
             if !method.is_allowed() {
```

Comment:
When the changes above are done then move the whole "metrics endpoint check if block" to line 215. (before the CORS check)